amazon-kinesis-client/src/main/java/software/amazon/kinesis/leases/dynamodb/DynamoDBLeaseManagementFactory.java

```diff
@@ -515,6 +516,29 @@ public ShardSyncTaskManager createShardSyncTaskManager(MetricsFactory metricsFac
                 metricsFactory);
     }
 
+    /**
+     * Create ShardSyncTaskManager from the streamConfig passed
+     *
+     * @param metricsFactory - factory to get metrics object
+     * @param streamConfig - streamConfig for which ShardSyncTaskManager needs to be created
+     * @return ShardSyncTaskManager
+     */
+    @Override
+    public ShardSyncTaskManager createShardSyncTaskManager(MetricsFactory metricsFactory, StreamConfig streamConfig,
```

Comment:
Should we chain this method with the one above since we handle null in the `HierarchicalShardSyncer` already?

Reply:
yes sounds like good idea. Done